### ISSUE TITLE
Inverted channel volume slides

### ISSUE THREAD

**Issue by amaennel** [michaelforney/oscmix#6]
Hi!

First of all great project, thanks for working on this.

While playing around with the web frontend I noticed a small issue with the channel volume sliders. 

When using firefox 123, which supports CSS vertical writing on range elements, the sliders have the wrong orientation. 
When using firefox 116 (which is the current snap version on ubuntu 22.04) , which doesn't support CSS vertical writing on range elements, the sliders have the expected orientation. 

I assume you tried adding `writing-mode: vertical-lr` to the sliders in a version that does not yet support this, but left it in accidentally. This css setting can just be removed I think.

See: https://developer.chrome.com/blog/vertical-form-controls

Another thing I have noticed is that on firefox 116 the channel settings, eq and dynamics do not collapse as they should. The css operator `:not` was only introduced in firefox 121, so this doesn't work for a vanilla ubuntu installation.

Long story short, I think a minimum browser version requirement should be in the readme, because this sadly is a reality for web dev. I can also make a pull request for these changes if you want. 

Cheers,
Alex

**Comment by amaennel** [comment]
> I'm not sure about 116, but if there's an easy way to display properly there, too, I'd like to support it.

I can't really think of an easier way, but this is not something I have looked into much. My intention with the issue was also not to have every browser compatibility issue fixed (impossible) but rather to make you aware of it. As there is currently a lot more important stuff to do, I still think that the best way to navigate this is to just provide a `Tested in chrome X.X` section in the README for now. 

> Hmm, ok. I think a lot of the :not logic could be inverted by changing the default to display: none, and setting it to the proper value (not sure what, maybe revert or something?) when the panel button is selected. Thoughts on that?

Default value for `div` is `display: block`.  But that will probably not solve it for [Firefox < 121](https://caniuse.com/?search=%3Ahas). Again not that it's super important. 

> Also, since you seem to be more web-savvy than me, maybe you could help with one issue I just couldn't seem to figure out: when you shrink the window vertically, the channel strips start to individually scroll vertically. This is because the volume range seems to have some minimum height. If I set it to height: 100px or something, the channels show up as I intend, but then of course its height is fixed. I can't figure out how to make them fill up the entire height of the channel channel, and shrink down further than their default height.

I'm not entirely sure why, but adding `min-height: 0` to `.channel-volume` solves this. `display: flex` sets `min-height: auto` on the flex-items per default so that might have something to do with it. Note that the minimum height of the channels is now implicitly set by `min-height: 200px` in `.channel-strip` so they still won't shrink completely.

A bit unrelated maybe: I feel like the WebMidi part isn't getting enough attention in the README. It's pretty cool that you got that working that well, because it allows people to use oscmix even if they aren't particularly linux-savvy. But I only found out that that exists via the forum thread.

**Comment by michaelforney** [comment]
> I'm not entirely sure why, but adding `min-height: 0` to `.channel-volume` solves this. `display: flex` sets `min-height: auto` on the flex-items per default so that might have something to do with it. Note that the minimum height of the channels is now implicitly set by `min-height: 200px` in `.channel-strip` so they still won't shrink completely.

Thank you! That did the trick.

> A bit unrelated maybe: I feel like the WebMidi part isn't getting enough attention in the README. It's pretty cool that you got that working that well, because it allows people to use oscmix even if they aren't particularly linux-savvy. But I only found out that that exists via the forum thread.

I don't really know what to add. Feel free to send a PR if you have some idea.

Closing this, since I believe the original should be fixed with new firefox.
